server/common/nosql/mongo-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/mongo/dao/AbstractMongoDao.java

```diff
@@ -107,50 +102,37 @@ protected long count(Query query) {
         return mongoTemplate.count(query, getDocumentClass());
     }
 
-    @Override
     public T save(T dto) {
         mongoTemplate.save(dto);
         return dto;
     }
 
-    @Override
-    public T persist(T dto) {
-        return save(dto);
-    }
-
-    @Override
-    public <V> V save(V dto, Class<V> clazz) {
+    public <V> V save(V dto, Class<?> clazz) {
         LOG.debug("Save entity of {} class", clazz.getName());
         mongoTemplate.save(dto);
         return dto;
     }
 
-    @Override
     public List<T> find() {
         LOG.debug("Find  all documents from [{}] collection.", getCollectionName());
         return mongoTemplate.findAll(getDocumentClass());
     }
 
-    @Override
-    public T findById(String id) {
-        return findById(id, false);
-    }
-
-    @Override
-    public T findById(String id, boolean lazy) {
-        LOG.debug("Find document of collection [{}] by id [{}]", getCollectionName(), id);
-        return mongoTemplate.findById(id, getDocumentClass());
+    public T findById(K key) {
+        LOG.debug("Find document of collection [{}] by id [{}]", getCollectionName(), key);
```

Comment:
I propose to use one of a log pattern, either "Find..." or "Try to find...", but not both.